validator/keymanager/v2/direct/direct.go

```diff
@@ -467,7 +490,90 @@ func (dr *Keymanager) createAccountsKeystore(
 	}, nil
 }
 
-func (dr *Keymanager) askUntilPasswordConfirms(
+func (dr *Keymanager) listenForFileChanges(ctx context.Context, filePath string) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.WithError(err).Error("Could not initialize file watcher")
+		return
+	}
+	defer func() {
+		if err := watcher.Close(); err != nil {
+			log.WithError(err).Error("Could not close file watcher")
+			return
+		}
+	}()
+	if err := watcher.Add(filePath); err != nil {
+		log.WithError(err).Errorf("Could not add file %s to file watcher", filePath)
+		return
+	}
+	for {
+		select {
+		case event := <-watcher.Events:
+			// If a file was modified, we attempt to read that file
+			// and parse it into our accounts store.
+			if event.Op&fsnotify.Write == fsnotify.Write {
+				fileBytes, err := ioutil.ReadFile(event.Name)
+				if err != nil {
+					log.WithError(err).Errorf("Could not read file at path: %s", event.Name)
+					continue
+				}
+				accountsKeystore := &v2keymanager.Keystore{}
+				if err := json.Unmarshal(fileBytes, accountsKeystore); err != nil {
+					log.WithError(
+						err,
+					).Errorf("Could not read valid, EIP-2335 keystore json file at path: %s", event.Name)
+					continue
+				}
+				if err := dr.reloadAccountsFromKeystore(accountsKeystore); err != nil {
+					log.WithError(
+						err,
+					).Error("Could not replace the accounts store from keystore file")
+				}
+			}
+		case err := <-watcher.Errors:
+			log.WithError(err).Errorf("Could not watch for file changes for: %s", filePath)
+		case <-ctx.Done():
+			return
+		}
+	}
+}
+
+// Replaces the accounts store struct in the direct keymanager with
+// the contents of a keystore file by decrypting it with the accounts password.
+func (dr *Keymanager) reloadAccountsFromKeystore(keystore *v2keymanager.Keystore) error {
+	// If no one is subscribing to account changes, no need to proceed.
```

Comment:
Why is it not required to incorporate the changes even if nobody is listening for change events?  If someone then subscribes, they won't get the latest.  This may be valid but makes assumptions about how Keymanager is used.